### ISSUE TITLE
Bootcamp adventures

### ISSUE THREAD

**Issue by blahah** [Blahah/mozilla_science_fellowship#25]
A sprint project, in which I turn https://github.com/Blahah/command_line_bootcamp into a modular system for browser-based teaching of any command-line-based materials (e.g. bioinformatics stuff, data science, devops, server management, etc...).

This is my initial rough plan...
### Seed ideas:
- Make it easy for people to deploy this system with their own materials and docker image - perhaps a generator, or pre-built configurable images for cloud providers.
- Make a version for the classroom, features like:
  - system runs on a local machine and serves to the local network
  - the teacher gets an interface that allows them to demo on screen, but also see how each student is progressing
-  Q&A - test questions can be set for each module and the user can't progress until they answer the questions correctly
- Environment syncs with module - when the user navigates to a particular module, the command-line environment gets set up for that module
- Reset environment - if the user makes a mess and wants to restore a clean environment, let them do that easily
### Process:

> research > design > implementation > user testing
#### Research
- Reach out to people who do computational teaching
- Ask for reviews of existing bootcamp example
- Discuss how they would want to use it
- Maybe design a survey to solicit initial feedback and then seek more detailed discussion
#### Design
- Persona exercise
- Route mapping exercise

**Comment by joeyklee** [comment]
Very cool - I think would be great is maybe some demo materials + a video to explain why and how command line knowledge is an important skill. Also, as we saw in the P5.js "hello" video - it is welcoming for users that might not immediately understand how cool it actually is to be able to automagically change 10000 file names using the command line, etc.

**Comment by blahah** [comment]
Ooh that's a nice idea! Like a showcase of what cool stuff you can do from the command-line. I'm now wondering if there could be a clihub or someting, a site where you can pick any community-provided tutorial for learning something at the command line, and launch it in your browser...

**Comment by joeyklee** [comment]
Yes! It would be great to use the "remix" idea like the Mozilla Thimble or to have like a roadmap of commandline adventures

**Comment by blahah** [comment]
Planning a small, painless survey to collect community feedback about which features are useful or which frustrations they feel with current tools: https://public.etherpad-mozilla.org/p/commandlinetut. Thanks to @stephwright for dropping science on me about how to make awesome surveys.

**Comment by stephwright** [comment]
Was thinking abt the idea of having some small group review your survey for usability / readability and realized you have a built-in group here that would a) be likely to respond, b) respond quickly, and c) wouldn't draw from your pool of likely survey respondents. :)

**Comment by blahah** [comment]
Excellent point!
